client-sdk/go/modules/contracts/types.go

```diff
@@ -179,6 +180,69 @@ type InstanceStorageQueryResult struct {
 	Value []byte `json:"value"`
 }
 
+type StoreKind uint32
+
+// String returns human-readable representation of StoreKind.
+func (sk StoreKind) String() string {
+	switch sk {
+	case StoreKindPublic:
+		return "public"
+	case StoreKindConfidential:
+		return "confidential"
+	}
+
+	return ""
+}
+
+// Unmarshal converts string to StoreKind.
+func (sk *StoreKind) Unmarshal(s string) error {
```

Comment:
The standard way to implement these is [`MarshalText`](https://pkg.go.dev/encoding#TextMarshaler)/[`UnmarshalText`](https://pkg.go.dev/encoding#TextUnmarshaler) (see examples already in the codebase).